server_admin/topics/admin-cli.adoc

```diff
@@ -1,1320 +1,1702 @@
-    $ kcadm.sh create users -s username=testuser -s enabled=true -r demorealm
-
-In this example we first start a session authenticated as `admin` user in `master` realm. Then we perform a POST call against the following
-resource URL:
-
-    http://localhost:8080/auth/admin/realms/demorealm/users
+This method updates the resource on the server with all the attributes in the sent JSON document.
 
+Another option is to perform an on-the-fly update using the [command]`-s, --set` options to set new values.
 
-Commands `create` and `update` by default send JSON body to the server. You can use `-f FILENAME` to read a pre-made document from a file.
-You can use `-f -`, and message body will be read from standard input.
-But you can also specify individual attributes and their values as seen in the previous `create users` example, and they will be composed into a JSON body and sent to the server.
-
-
-When using `update` there are several ways to update a resource. You can first get current state of a resource, and save it into a file,
-then edit that file and send it to the server for update. For example:
-
-    $ kcadm.sh get realms/demorealm > demorealm.json
-    $ vi demorealm.json
-    $ kcadm.sh update realms/demorealm -f demorealm.json
-
-This way the resource on the server will be updated with all the attributes in the sent JSON document.
-
-Another option is to perform an update on-the-fly using `-s, --set` options to set new values. For example:
-
-    $ kcadm.sh update realms/demorealm -s enabled=false
+For example:
+[options="nowraps"]
+----
+$ kcadm.sh update realms/demorealm -s enabled=false
+----
 
-That would only update `enabled` attribute to `false`.
+That method only updates the [command]`enabled` attribute to `false`.
 
-By default `update` operation first performs a `get`, and then merges new attribute values with existing.
-Mostly this is a preferred behaviour. In some cases the endpoint may support `PUT` but not `GET`.
-You can use `-n` option to perform a so called 'no-merge' update which performs a PUT, without first doing a GET.
+By default, the [commamd]`update` command first performs a [command]`get` and then merges the new attribute values with existing values. This is the preferred behavior. In some cases, the endpoint may support the [command]`PUT` command but not the [command]`GET` command. You can use the [command]`-n` option to perform a "no-merge" update, which performs a [command]`PUT` command without first running a [command]`GET` command.
 
 
 === Realm operations
 
-Creating a new realm::
-
-To create a new enabled realm use `create` operation on `realms` endpoint, and set attributes `realm` and `enabled`:
-
-    $ kcadm.sh create realms -s realm=demorealm -s enabled=true
-
-Realm is not enabled by default. By enabling it, it can be used for authentication immediately.
-
-A description for a new object can be in JSON format as well:
+==== Creating a new realm
```

Comment:
It is always tough to maintain that delicate balance between aesthetics and usefulness. The hard question that I can't yet answer for this project is whether our users will know to look in the '===Realm operations' section to find the info they need or whether they need the sign posts provided by a heading listed in the sidebar.

In the short term, I will defer to @mstruk's opinion here, since he is much more familiar with this project and I'm the newb. :)

Reply:
Don't get me wrong - I know that using labels instead of subtitles is a hack. The real fix is to have a config setting that allows me to control to what depth subtitles should be included in navigation, and allow such control on a per-chapter basis. But I think we first need to find this fix before we can merge this change of labels to subtitles.

Reply:
👍

Reply:
@mstruk Is this a possible solution?  https://stackoverflow.com/questions/34481638/how-to-use-tocify-with-asciidoctor-for-a-dynamic-toc

Reply:
Here is another option.

If the title text should not be numbered or appear in the table of contents, use a heading at the appropriate level, and then set the discrete attribute above the heading.

[discrete]
==== Delete a realm
Text text text...

Reply:
Fantastic, that's it!
I tried it and it works. Do you want to add that to all fourth level subtitles in admin-cli.adoc?

Reply:
👍

Reply:
@stianst , who do we need to talk to to implement this solution that @leswilliams44 found above and @mstruk has tested? The solution is being suggested because this is a very long page that has too many entries in the table of contents. The ultimate solution would be for me to break this up into multiple pages, but that isn't something I have bandwidth for today (but it is on my TODO list for the future).

https://stackoverflow.com/questions/34481638/how-to-use-tocify-with-asciidoctor-for-a-dynamic-toc

Essentially, it involves adding a js file (full source in the link above) and then running asciidoctor with different options to build.

So, it looks like a great solution for asciidoctor, but I don't know how or even if this is feasible with our current toolchain. If so, I would like us to consider it. If not, I wonder if there is another option.

Do you know, and if not, do you know who I should be asking? Thanks.

Reply:
@matthewhelmke The [discrete] solution doesn't require anything extra in the build. It is supported out of the box. All that has to be done is to insert [discrete]  in front of all fourth level subtitles in admin-cli.adoc.

Reply:
@mstruk OH! I totally misread the earlier comment. Yes! Let's just use [discrete]. That is elegant and simple. Thank you for correcting my understanding.

Reply:
I will apply the [discrete] attribute to all level four headings.